test/jdk/java/awt/Frame/InitialIconifiedTest.java

```diff
@@ -50,36 +50,75 @@ public class InitialIconifiedTest {
 
     private static Robot robot;
 
+    private static final StringBuilder FAILURES = new StringBuilder();
+
     public static void main(String[] args) throws Exception {
         robot = new Robot();
-
         try {
-            EventQueue.invokeAndWait(InitialIconifiedTest::initAndShowGui);
+            EventQueue.invokeAndWait(InitialIconifiedTest::initAndShowBackground);
             robot.waitForIdle();
             robot.delay(500);
-            test();
+
+            test(false);
+            test(true);
         } finally {
             EventQueue.invokeAndWait(() -> {
                 backgroundFrame.dispose();
                 testedFrame.dispose();
             });
         }
+
+        if (!FAILURES.toString().isBlank()) {
+            throw new RuntimeException(FAILURES.toString());
+        }
+    }
+
+    private static void test(boolean isUndecorated) throws Exception {
+        String prefix = isUndecorated ? "undecorated" : "decorated";
+
+        EventQueue.invokeAndWait(() -> initAndShowTestedFrame(isUndecorated));
+        // On macos, we can observe the animation of the window from the initial
+        // NORMAL state to the ICONIFIED state,
+        // even if the window was created in the ICONIFIED state.
+        // The following delay is commented out to capture this animation
+        // robot.waitForIdle();
+        // robot.delay(500);
```

Comment:
Tested on macOS and windows. 
The test works on macOS with the work-around (extra delay).